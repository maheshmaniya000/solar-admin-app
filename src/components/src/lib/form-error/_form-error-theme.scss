@use '~@angular/material' as mat;

@mixin theme($theme-or-color-config) {
  $typography-config: mat.get-typography-config($theme-or-color-config);

  mat-error {
    @include mat.typography-level($typography-config, caption);
  }
}
