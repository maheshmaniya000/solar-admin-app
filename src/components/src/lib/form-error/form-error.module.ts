import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MatFormFieldModule } from '@angular/material/form-field';
import { TranslocoModule } from '@ngneat/transloco';

import { FormErrorComponent } from './append-form-error/form-error-component/form-error.component';
import { AppendFormErrorToCheckboxDirective } from './append-form-error/to-form-check-box/append-form-error-to-checkbox.directive';
import { AppendFormErrorToFormFieldDirective } from './append-form-error/to-form-form-field/append-form-error-to-form-field.directive';
import { AppendFormErrorToFormGroupDirective } from './append-form-error/to-form-group/append-form-error-to-form-group.directive';

@NgModule({
  declarations: [
    AppendFormErrorToFormFieldDirective,
    AppendFormErrorToCheckboxDirective,
    AppendFormErrorToFormGroupDirective,
    FormErrorComponent
  ],
  imports: [CommonModule, TranslocoModule, MatFormFieldModule],
  exports: [
    AppendFormErrorToFormFieldDirective,
    AppendFormErrorToCheckboxDirective,
    AppendFormErrorToFormGroupDirective
  ]
})
export class FormErrorModule {}
