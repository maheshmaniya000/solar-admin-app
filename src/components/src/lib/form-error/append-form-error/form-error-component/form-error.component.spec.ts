import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';

import { FormErrorComponent } from './form-error.component';

describe('FormErrorComponent', () => {
  describe('component', () => {
    const createComponent = createComponentFactory({
      component: FormErrorComponent,
      shallow: true
    });
    let spectator: Spectator<FormErrorComponent>;

    beforeEach(() => {
      spectator = createComponent({
        props: {
          errorMessage: 'some error',
          className: 'some-class'
        }
      });
    });

    it('should match snapshot with bound attributes', () => {
      expect(spectator.fixture).toMatchSnapshot();
    });
  });
});
