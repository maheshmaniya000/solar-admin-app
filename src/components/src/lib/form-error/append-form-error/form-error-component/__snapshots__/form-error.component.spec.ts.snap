// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormErrorComponent component should match snapshot with bound attributes 1`] = `
<sg-form-error
  __ngContext__={[Function LRootView]}
  className={[Function String]}
  errorMessage={[Function String]}
>
  <mat-error
    class="some-class"
  >
     some error 
  </mat-error>
</sg-form-error>
`;
